torch/xpu/streams.py

```diff
@@ -96,6 +96,28 @@ def __repr__(self):
         return f"torch.xpu.Stream(device={self.device} sycl_queue={self.sycl_queue:#x})"
 
 
+class ExternalStream(Stream):
```

Comment:
Why is this one a cleass and not just a function that creates this very specific Stream object? More or less just a binding to at::xpu::getStreamFromExternal

Reply:
Actually, here we follow CUDA's practice. I guess the new class `ExternalStream` could help the user to determine if a stream object is an external stream via `type(stream) == torch.xxx.ExternalStream`. And remind the user to be aware of the limitations of external streams like life cycle issues which is different from the native stream.

If you prefer a new binding function instead of the new class, I could try to re-design it.

Reply:
@albanD I add `torch.xxx.get_stream_from_external` API for CUDA and XPU. I don't think each kind of accelerator has the concept of the external stream. For mps, it seems like only have a stream. So I don't add this API to torch.accelerator namespace. What do you think about it?